modules/codegen/src/smithy4s/codegen/internals/Renderer.scala

```diff
@@ -990,9 +984,9 @@ private[internals] class Renderer(compilationUnit: CompilationUnit) { self =>
         (bd: BigDecimal) => line"scala.math.BigDecimal($bd)"
       case Primitive.BigInteger => (bi: BigInt) => line"scala.math.BigInt($bi)"
       case Primitive.Unit       => _ => line"()"
-      case Primitive.Double     => t => line"${t.toString}"
-      case Primitive.Float      => t => line"${t.toString}"
-      case Primitive.Long       => t => line"${t.toString}"
+      case Primitive.Double     => t => line"${t.toString}d"
+      case Primitive.Float      => t => line"${t.toString}f"
+      case Primitive.Long       => t => line"${t.toString}L"
```

Comment:
minor: any reason to use different casing on the suffix? 1L, 1.0F and 1.0D

I guess D looking alike 0 is one of them¿

Reply:
you actually get a compiler warning if you use lowercase `L`. The rest, the compiler doesn't care. That's the reason.